src/proto.c

```diff
@@ -558,16 +558,16 @@ int objectDataWrite(Object* obj, File* stream)
                 if (fileWriteInt32(stream, data->scenery.stairs.field_0) == -1) return -1;
                 break;
             case SCENERY_TYPE_ELEVATOR:
-                if (fileWriteInt32(stream, data->scenery.elevator.field_0) == -1) return -1;
-                if (fileWriteInt32(stream, data->scenery.elevator.field_4) == -1) return -1;
+                if (fileWriteInt32(stream, data->scenery.elevator.type) == -1) return -1;
+                if (fileWriteInt32(stream, data->scenery.elevator.level) == -1) return -1;
                 break;
             case SCENERY_TYPE_LADDER_UP:
                 if (fileWriteInt32(stream, data->scenery.ladder.field_0) == -1) return -1;
-                if (fileWriteInt32(stream, data->scenery.elevator.field_4) == -1) return -1;
+                if (fileWriteInt32(stream, data->scenery.elevator.level) == -1) return -1;
                 break;
             case SCENERY_TYPE_LADDER_DOWN:
                 if (fileWriteInt32(stream, data->scenery.ladder.field_0) == -1) return -1;
-                if (fileWriteInt32(stream, data->scenery.elevator.field_4) == -1) return -1;
+                if (fileWriteInt32(stream, data->scenery.elevator.level) == -1) return -1;
```

Comment:
@alexbatalov this looks like a bug. Shouldn't it be ladder instead of elevator?

Reply:
Yes, you're right. It does not break things because these fields are at the same offset.